### PR TITLE
Upgraded solution to .NET Core 3.1 and the latest libraries

### PR DESCRIPTION
**Pull request by josevarela**
Upgraded solution to .NET Core 3.1 and the latest libraries in this PR. A subsequent update to the solution will move it to the v2 endpoint of the Microsoft Identity Platform and the MSAL client library.

### Upgraded solution to .NET Core 3.1

- All projects upgraded to .NET Core 3.1
- Rearranged app initialization in program.cs and startup.cs
- Minor updates to a fix a few deprecated Apis and breaking changes
- Minor updates to the OpenId security code to deal with breaking changes while still using v1 of the Api and the ADAL client library.

### Upgraded JavaScript build

- Gulp and bower versions and npm packages used in the build
- Gulp tasks are now async

### Other updates

- Removed references to AddMemoryCache in the startup.cs file of the web project. The call is not required for the sample application to work

### CODE REVIEW

File: src/Tailspin.Surveys.Web/Controllers/SurveyController.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public async Task<IActionResult> Index()
                 var userId = User.GetSurveyUserIdValue();
                 var user = User.GetObjectIdentifierValue();
                 var issuerValue = User.GetIssuerValue();
-                var actionName = ControllerContext.ActionDescriptor.DisplayName;
+                var actionName = $"{typeof(SurveyController).FullName}.{nameof(Index)}";
```
Comment (kirpasingh) [inline src/Tailspin.Surveys.Web/Controllers/SurveyController.cs:62]: I must be missing something here but ```DisplayName``` property is available in [Asp.NET Core](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.abstractions.actiondescriptor.displayname?view=aspnetcore-3.1) - is there a particular reason to use different approach here?

Comment (josevarela) [reply src/Tailspin.Surveys.Web/Controllers/SurveyController.cs:62]: True. This change was made to accommodate a failing (previously-existing) unit test. Do you feel is important the action descriptor property for logging?

Comment (kirpasingh) [reply src/Tailspin.Surveys.Web/Controllers/SurveyController.cs:62]: As long as everything put together is working, I don't really care! Thank you for clarifying.

**Comment by josevarela** [comment]
@kirpasingh The solution can be run locally but there is actually a bit of configuration required before can run it via F5. The steps are documented in this page:[https://github.com/mspnp/multitenant-saas-guidance/blob/master/get-started.md]( https://github.com/mspnp/multitenant-saas-guidance/blob/master/get-started.md), which is linked from the root readme.md  of the repo

**Pull request merged**